src/stash/src/postgres.rs

```diff
@@ -868,6 +896,56 @@ impl Stash {
         }
         Ok(version == committed_if_version)
     }
+
+    #[tracing::instrument(name = "stash::is_initialized", level = "debug", skip_all)]
+    pub async fn is_initialized(&mut self) -> Result<bool, StashError> {
+        let initialized =
+            COLLECTION_CONFIG.upper(self).await?.elements() != [crate::Timestamp::MIN];
```

Comment:
Although I think this generally works, I'm a bit scared that we have lots of code that will append empty batches during peek (that is, many reads can also cause writes), and could potentially mess this up? Can we do something else like assert that `iter[_raw]` over this collection is empty? That seems safer than relying on a specific timestamp.

Reply:
Makes sense. I updated the impl to first read what collections we have (at a given point the Stash's `collections` member var might not be initialized) and only if the config collection exists do we try to read it. I think this should avoid any writes from occurring? If I missing something though let me know!